### DISCUSSION TITLE
JavaScript plugin crashes/disappears randomly. How to debug?

### DISCUSSION THREAD

**Discussion by TigerHix** [elgatosf/elgato#50] [Help]
Hello, this problem has bothered me for a while. I am developing a JavaScript plugin based on the official PropertyInspector template, and the plugin itself works just fine, except every once in a while, the plugin no longer works (i.e. pressing on any actions gives the yellow exclamation mark triangle icon). When I click on the action, I can see the property inspector page still be able to loaded, but it cannot communicate with the plugin - which is made evident by opening localhost:23654 and not seeing my plugin's index.html.

Every time this happens, I can restart Stream Deck and my plugin starts working again (and is visible on localhost:23654). But I don't see other plugins suffering from the same problem, so I really wonder if there is a way to address this. I have looked into the log folder `%appdata%\Elgato\StreamDeck\logs\` but there's no log entries relevant to my plugin or even any that contains keywords like "crash".

Any help would be appreciated. Thanks!

**Comment by tiptronic** [comment]
Hi @TigerHix 

did you check other debugging options as shown here:

https://docs.elgato.com/sdk/plugins/getting-started#5.-debugging

?
